model/src/resource/device.rs

```diff
@@ -75,30 +87,64 @@ impl Device {
     #[must_use]
     pub fn to_json(&self, api: &APIVersion) -> DeviceJson {
         match *api {
-            V1_0 => {
-                // Senders
-                let senders = self.senders.iter().map(ToString::to_string).collect();
-
-                // Receivers
-                let receivers = self.receivers.iter().map(ToString::to_string).collect();
-
-                DeviceJson::V1_0(is_04::v1_0_x::Device {
-                    id: self.core.id.to_string(),
-                    version: self.core.version.to_string(),
-                    label: self.core.label.clone(),
-                    type_: self.type_.to_string(),
-                    node_id: self.node_id.to_string(),
-                    senders,
-                    receivers,
-                })
-            }
+            V1_0 => DeviceJson::V1_0((*self).clone().into()),
+            V1_3 => DeviceJson::V1_3((*self).clone().into()),
             _ => panic!("Unsupported API"),
         }
     }
 }
 
+impl Registerable for Device {
+    fn registry_path(&self) -> String {
+        format!("devices/{}", self.core.id)
+    }
+
+    fn registration_request(&self, api: &APIVersion) -> serde_json::Value {
+        match self.to_json(api) {
+            DeviceJson::V1_0(json) => registration_request!(json, v1_0_x),
+            DeviceJson::V1_3(json) => registration_request!(json, v1_3_x),
+        }
+    }
+}
+
 #[derive(Debug, Serialize)]
 #[serde(untagged)]
 pub enum DeviceJson {
-    V1_0(is_04::v1_0_x::Device),
+    V1_0(v1_0_x::Device),
+    V1_3(v1_3_x::Device),
```

Comment:
You could just add a API version here and implement Into<...> for it in the future. For now i just did this to implement v1.3 without breaking the v1.0 implementation

This was done for all resources.